python/cudf/cudf/core/dataframe.py

```diff
@@ -1881,23 +1881,45 @@ def pop(self, item):
         del self[item]
         return popped
 
-    def rename(self, mapper=None, columns=None, copy=True, inplace=False):
-        """
-        Alter column labels.
+    def rename(
+        self,
+        mapper=None,
+        columns=None,
+        index=None,
+        copy=True,
+        inplace=False,
+        axis=0,
```

Comment:
Reorder these to follow Pandas order: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.rename.html

And add all of the kwargs in that function to align the APIs. If we don't support anything yet check that it's not the default value and throw a `NotImplementedError` accordingly.